### PR TITLE
Add an invoke task to create and destroy a vm

### PR DESCRIPTION
**Pull request by ogaca-dd**
What does this PR do?
---------------------

Add an invoke task to create and destroy a vm.
Note: More options are going to be added to configure a vm in a later PR like for configuring the operating system and the architecture.

### CODE REVIEW

**Review by pducolin — APPROVED** [review]
I like how this will help discover environments and options !

Have you considered using `golang` task managers ? I can see how agent teams are used to `invoke` so ok for using it, I remember investigating about go solutions in the past and there are good alternatives, just in case we want to keep this 100% golang

File: tasks/vm.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    aws_deploy(
+        config,
+        stack_vm,
+        install_agent,
+        f"ddagent:deploy={install_agent}",
+        f"ddagent:version={agent_version}",
+    )
+
+
+def aws_deploy(config, scenario_name, install_agent, *args):
```
Comment (vboulineau) [inline tasks/vm.py:30]: I think this belongs to another file

File: tasks/vm.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if api_key is None or len(api_key) != 32:
+            raise invoke.Exit("Invalid API KEY. You must define the environment variable 'DD_API_KEY'. " +
+                              "If you don't want an agent installation add '--no-install-agent'.")
+
+    scenario_folder = get_scenario_folder(scenario_name)
+    stack_name = get_stack_name(scenario_name)
+    cmd_args = [
+        "aws-vault", "exec", "sandbox-account-admin", "--",
+        "pulumi", "up",
+        "-c", "ddinfra:aws/defaultKeyPairName=" + config["aws_key_pair"],
```
Comment (vboulineau) [inline tasks/vm.py:43]: The way I would see it is a `ParameterBuilder` that generates option based on `Configuration` dataclass.

File: tasks/vm.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    if install_agent:
+        api_key = os.getenv("DD_API_KEY")
+        if api_key is None or len(api_key) != 32:
+            raise invoke.Exit("Invalid API KEY. You must define the environment variable 'DD_API_KEY'. " +
+                              "If you don't want an agent installation add '--no-install-agent'.")
+
+    scenario_folder = get_scenario_folder(scenario_name)
+    stack_name = get_stack_name(scenario_name)
+    cmd_args = [
+        "aws-vault", "exec", "sandbox-account-admin", "--",
```
Comment (vboulineau) [inline tasks/vm.py:41]: I don't think we can really enforce that. It does not on all setups (for instance, I use a `ServiceAccount`), it also does not work in CI if we implement `stack as a service`.

I think we need some run profiles  to handle that. The wrapping logic (i.e. running with `aws-vault`), should be separated from the construction of the parameters and Pulumi commands.

**Review by vboulineau — COMMENTED** [review]
Also requires a `requirements.txt` file.

File: tasks/stack.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+def get_stack_name(scenario_name):
+    stack_name = stacks[scenario_name][1]
+    return getpass.getuser() + "-" + stack_name
+
+
+def get_scenario_folder(scenario_name):
+    return stacks[scenario_name][0]
+
+
+def get_scenario_folder_from_stack_name(full_stack_name):
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply tasks/stack.py:17]: >. This methods is perhaps not entirely reliable.
Why?

File: tasks/vm.py (hunk 1 above)
Comment (ogaca-dd) [reply tasks/vm.py:30]: I going to move it to another file when adding new command in a later PR.

**Comment by ogaca-dd** [comment]
Closed in favor of https://github.com/DataDog/test-infra-definitions/pull/68

**Pull request closed without merging**